### ISSUE TITLE
Epic: Host Open VSX mirror for managed Gitpod

### ISSUE THREAD

**Issue by hm2075** [gitpod-io/gitpod#11716]
Host Open VSX within Gitpod with the intent to improve reliability of Gitpod. 

**Please note:** Not to be confused with self-hosting Open VSX for Gitpod customers, that work is encapsulated in: 

**Remaining**
- [ ] Documentation of Open VSX mirror
- [ ] We don't have a process for automatic deployments (manual deployment)
- [ ] Contribute our changes back upstream https://github.com/eclipse/openvsx/pull/586

**Completed**
- [x] [Announce changes](https://www.gitpod.io/changelog#reliable-vs-code-extensions-gitpod-now-hosting-an-open-vsx-mirror)
- [x] log exceptions as errors in GCP
- [x] werft job does not redeploy if only charts are changed
- [x] server has to be restarted if application.yaml is changes, they should either hot load or do rolling deployment 

**Related issues**
- https://github.com/gitpod-io/gitpod/issues/12004

<details>
<summary>Original description</summary>
WIth open-vsx going being unavailable for extended periods gitpod should stop relying on that server
they have a poor resolution times and it can be days until issues are resolved

other options should be considered such as saving extensions locally to gitpod servers
</details>

**Comment by akosyakov** [comment]
We are looking into POC for self-hosting OpenVSX this week.

**Comment by loujaybee** [comment]
Related epic: 

- https://github.com/gitpod-io/gitpod/issues/12004

**Comment by akosyakov** [comment]
We are going to investigate how we deploy it our infrastructure.

**Comment by loujaybee** [comment]
We're blocked on this one based on some upstream Open VSX issues @akosyakov is going to chase those up and we'll pick this up when some of those issues are resolved.

**Comment by loujaybee** [comment]
It seems that we're pretty much complete with this epic, streamlining the deployment / operations could be part of a separate issue or epic.

**Comment by loujaybee** [comment]
Some remaining work, but can be completed in separate issues / epic: 

* Documentation of Open VSX mirror
* We don't have a process for automatic deployments (manual deployment)
* Contribute our changes back upstream

Closing 🚀
